src/Processors/Processor.cs

```diff
@@ -0,0 +1,62 @@
+﻿namespace Processors;
+
+public abstract class Processor : BaseThread
+{
+    private bool _isPaused = false;
+    private bool _onPauseOnce = false;

```

Comment:
Why `Once`?

Reply:
I don't think you even need this variable if you have `_isPaused`

Reply:
It's a variable to decide when to trigger the `OnPause()` method just once and not every time into the `While` loop after I invoke the `PauseRequest()` method of the process instance.

Reply:
I reworked this class. Please check the updated version.